boards/xtensa/intel_adsp_cavs15/tools/adsplog.py

```diff
@@ -40,31 +40,39 @@
 WIN_SIZE = 0x20000
 LOG_OFFSET = WIN_OFFSET + WIN_IDX * WIN_SIZE
 
-# List of known ADSP devices by their PCI IDs
-DEVICES = ["8086:5a98"]
-
 mem = None
+pcid="/sys/bus/pci/devices"
+
+for fname in os.listdir(pcid):
+    pfile = pcid + "/" + fname + "/class"
+    if not os.path.exists(pfile):
+        continue
+    fd = open(pfile)
+    pciclass = fd.read()
 
-for dev in DEVICES:
-    # Find me a way to do this detection as cleanly in python as shell, I
-    # dare you.
-    barfile = subprocess.Popen(["sh", "-c",
-                                "echo -n "
-                                "$(dirname "
-                                f"  $(fgrep PCI_ID={dev.upper()} "
-                                "    /sys/bus/pci/devices/*/uevent))"
-                                "/resource4"],
-                               stdout=subprocess.PIPE).stdout.read()
-    if not os.path.exists(barfile):
+    pfile = pcid + "/" + fname + "/vendor"
+    if not os.path.exists(pfile):
```

Comment:
vendor is mandatory too, drop this too. Let's keep this script smaller than the previous one :-)

Reply:
Ack.